pyunifiprotect/utils.py

```
@@ -219,3 +220,13 @@ def serialize_list(items: Iterable[Any]) -> List[Any]:
 def round_decimal(num: Union[int, float], digits: int) -> Decimal:
     """Rounds a decimal to a set precision"""
     return Decimal(str(round(num, digits)))
+
+
+def ip_from_host(host: str) -> IPv4Address:
+    try:
+        return IPv4Address(host)
+    except AddressValueError:
+        pass
+
+    _, _, ips = socket.gethostbyname_ex(host)
```

Can this fail with `socket.gaierror`.  Should we let it blow up?

---

I am thinking we actually should let that blow up. If the DNS name cannot resolve, the code will have other issues working.

---

Let me do some testing with bad data/bad network and see what the exception looks like. If it is confusing, we can catch it and make it easier to understand.

---

This is what I get from various errors:

```python
>>> socket.gethostbyname("google.com")  # timeout
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.gaierror: [Errno -3] Temporary failure in name resolution

>>> socket.gethostbyname("notarealdomain.com") # success
'64.90.40.98'

>>> socket.gethostbyname("notarealdomainsdf.com")  # bad DNS
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.gaierror: [Errno -2] Name or service not known
```

So these are pretty common errors I have seen all the time. I think this one is safe to let it go through. Should be errors people recognize.